### DISCUSSION TITLE
Future daily notes

### DISCUSSION THREAD

**Discussion by daviddumonde** [SilentVoid13/Templater#489] [Help]
I’m enjoying the great things I can do with templater. The only snag so far is when I click a future date on the calendar to create a daily note and templater still automatically inserts today’s date instead of the future date. It’s not a big deal to edit it to the correct date, but I wondered if there is a way to have it recognize that it’s creating a daily note for a future date and enter the correct date?

Thanks for a great plugin!

**Comment by I-Pch** [comment]
Hi @daviddumonde 😊 

To create future or passed daily notes using `Calendar` and having Templater display the correct date in the note you could use something like this : 
```javascript
<% tp.date.now("YYYY-MM-DD", 0, tp.file.title, "YYYY-MM-DD") %>
```

or even this : 
```javascript
<% moment(tp.file.title,'YYYY-MM-DD').format('YYYY-MM-DD') %>
```

Both use the date from the title of the daily note as a reference instead of `now`/`today` (in these cases, the title of the daily note is formatted as `YYYY-MM-DD`)

Hope this helps 😇

**Reply by abhinav4848** [reply]
This is what I'm using to make new daily notes as far into the future or the past as needed
```js
<< [[<% tp.date.now("YYYY-MM-DD", -1, tp.file.title, "YYYY-MM-DD") %>]] | [[<% tp.date.now("YYYY-MM-DD", 1, tp.file.title, "YYYY-MM-DD") %>]] >>
```

As mentioned above, this code relies on the filename (formatted as YYYY-MM-DD) to generate links to one day in the past and one day in the future
